hbase-server/src/main/java/org/apache/hadoop/hbase/master/SplitWALManager.java

```diff
@@ -184,4 +185,19 @@ public void releaseSplitWALWorker(ServerName worker, MasterProcedureScheduler sc
   public void addUsedSplitWALWorker(ServerName worker) {
     splitWorkerAssigner.addUsedWorker(worker);
   }
+
+  public String renameWALForRetry(String walPath, Integer workerChangeCount) throws IOException {
+    String originalWALPath;
+    if (workerChangeCount == 0) {
+      originalWALPath = walPath;
+    } else {
+      originalWALPath = walPath.substring(0, walPath.length() - RETRYING_EXT.length() - 3);
+    }
+    String walNewName =
```

Comment:
So when retrying number is 0, we also have the '.retrying' suffix? Will this cause trouble when upgrading?

Reply:
I do not think the current code reflection your explaination here...

If you do not want to change the wal name when retry count == 0, you should just return at the first if condition?

Reply:
We call this method in RELEASE_SPLIT_WORKER state. At this time **first try** of wal split is already complete. We only reach here if fir try is not able to split the wal.

Reply:
Then please add this comment as a javadoc of this method?

Reply:
done